mavros_extras/src/plugins/terrain.cpp

```diff
@@ -0,0 +1,74 @@
+/**
+ * @brief Terrain plugin
+ * @file terrain.cpp
+ * @author Matt Anderson <anderson_rayner@hotmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2021 Ardupilot.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <mavros_msgs/Terrain.h>
+
+namespace mavros {
+namespace extra_plugins {
+/**
+ * @brief Terrain height plugin.
+ *
+ * This plugin allows publishing of terrain height estimate from FCU to ROS.
+ *
+ */
+class TerrainPlugin : public plugin::PluginBase {
+public:
+	TerrainPlugin() : PluginBase(),
+		terrain_nh("~terrain")
+	{ }
+
+	void initialize(UAS &uas_) override
+	{
+		PluginBase::initialize(uas_);
+
+		terrain_pub = terrain_nh.advertise<mavros_msgs::Terrain>("in", 10);
+	}
+
+	Subscriptions get_subscriptions() override
+	{
+		return {
+			make_handler(&TerrainPlugin::handle_terrain)
+		};
+	}
+
+private:
+	ros::NodeHandle terrain_nh;
+
+	ros::Publisher terrain_pub;
+
+	void handle_terrain(const mavlink::mavlink_message_t *msg, mavlink::common::msg::TERRAIN_REPORT &terrain) {
```

Comment:
Method name is ok, but i i prefer `handle_{MESSAGE_NAME.lower()}`.

Reply:
Good suggestion, fixed!